R/SEBC-methods.R

```diff
@@ -0,0 +1,30 @@
+# Basic methods for the SummarizedExperimentByColumn class,
+# an internal class that powers the alt* suite of methods.
+
+.get_se <- function(x) x@se
+
+SummarizedExperimentByColumn <- function(se) new("SummarizedExperimentByColumn", se=se)
+
+setMethod("length", "SummarizedExperimentByColumn", function(x) ncol(.get_se(x)))
+
+setMethod("[", "SummarizedExperimentByColumn", function(x, i, j, ..., drop=FALSE) {
+    initialize(x, se=.get_se(x)[,i])
+})
+
+setReplaceMethod("[", "SummarizedExperimentByColumn", function(x, i, j, ..., value) {
+    left <- .get_se(x)
+    left[,i] <- .get_se(value)
+    initialize(x, se=left)
+})
+
+setMethod("c", "SummarizedExperimentByColumn", function(x, ...) {
```

Comment:
Is this what `c` does for a SummarizedExperiment?  I get 

> Error in validObject(.Object) : 
>   invalid class “SummarizedExperiment” object: 
>     nb of rows in 'assay' (200) must equal nb of rows in 'rowData' (400)

when I attempt to `c` two SummarizedExperiments together.

Reply:
No, `c` shouldn't do anything for SEs. There's no need for the SEBC to behave like an SE, it's just a wrapper around the SE to get us column subsetting/combining behavior.